### ISSUE TITLE
Override endpoint-url with CloudCustodian and policy testing

### ISSUE THREAD

**Issue by danieleb0101** [capitalone/cloud-custodian#2540]
Hi, I wrote a policy and I would like to test it.
Ideally I would like to do some mocking like Cloud Custodian does with placebo, but I cannot reuse anything unless I checkout the Cloud Custodian repo locally, and I refer to the path of zpill.py and common.py to reproduce something similar to https://github.com/capitalone/cloud-custodian/blob/master/tests/test_ec2.py . 
Definitely a cumbersome solution, unless I am missing something.
So, I am inclined to use a solution with python moto, this library allows to run a stand alone server which would simulate a dummy EC2 API endpoint, and then I can create my testing environment. So I would run my policy against this dummy environment and I would check if the results are as expected.
However, what I really need to make it work is to override EC2 API endpoint url. With CLI it is possible to do that using --endpoint-url (see https://docs.aws.amazon.com/cli/latest/reference/index.html#cli-aws). How can I override the endpoint url with Cloud Custodian? Is that possible?

**Comment by thisisshi** [comment]
This isn’t possible in custodian as of now, but that does sound like a great way to introduce a policy testing framework, I haven’t worked with overriding the aws endpoint or with Moto but that definitely sounds like a worthwhile addition imo

**Comment by danieleb0101** [comment]
Yes, another useful feature would be to have the option `--no-verify-ssl` as in the cli.
In general it would be great to have most of the options that are available in the cli ie. `--endpoint-url`,`--no-verify-ssl`,`--ca-bundle`, `--cli-read-timeout` and `--cli-connect-timeout`. Options in the cli are here: https://docs.aws.amazon.com/cli/latest/reference/index.html#options

**Comment by miki725** [comment]
`--endpoint-url` would help with testing cloudcustodian policies against localstack tremendously

https://docs.localstack.cloud/overview/
